shell/platform/windows/angle_surface_manager.cc

```diff
@@ -147,92 +116,176 @@ bool AngleSurfaceManager::Initialize(bool enable_impeller) {
   // Attempt to initialize ANGLE's renderer in order of: D3D11, D3D11 Feature
   // Level 9_3 and finally D3D11 WARP.
   for (auto config : display_attributes_configs) {
-    bool should_log = (config == display_attributes_configs.back());
-    if (InitializeEGL(egl_get_platform_display_EXT, config, should_log)) {
-      break;
+    bool is_last = (config == display_attributes_configs.back());
+
+    display_ = egl_get_platform_display_EXT(EGL_PLATFORM_ANGLE_ANGLE,
+                                            EGL_DEFAULT_DISPLAY, config);
+
+    if (display_ == EGL_NO_DISPLAY) {
+      if (is_last) {
+        LogEglError("Failed to get a compatible EGLdisplay");
+        return false;
+      }
+
+      // Try the next config.
+      continue;
     }
+
+    if (::eglInitialize(display_, nullptr, nullptr) == EGL_FALSE) {
+      if (is_last) {
+        LogEglError("Failed to initialize EGL via ANGLE");
+        return false;
+      }
+
+      // Try the next config.
+      continue;
+    }
+
+    return true;
   }
 
-  EGLint numConfigs = 0;
+  FML_UNREACHABLE();
+}
+
+bool AngleSurfaceManager::InitializeConfig(bool enable_impeller) {
+  const EGLint config_attributes[] = {EGL_RED_SIZE,   8, EGL_GREEN_SIZE,   8,
+                                      EGL_BLUE_SIZE,  8, EGL_ALPHA_SIZE,   8,
+                                      EGL_DEPTH_SIZE, 8, EGL_STENCIL_SIZE, 8,
+                                      EGL_NONE};
+
+  const EGLint impeller_config_attributes[] = {
+      EGL_RED_SIZE,       8, EGL_GREEN_SIZE, 8, EGL_BLUE_SIZE,    8,
+      EGL_ALPHA_SIZE,     8, EGL_DEPTH_SIZE, 0, EGL_STENCIL_SIZE, 8,
+      EGL_SAMPLE_BUFFERS, 1, EGL_SAMPLES,    4, EGL_NONE};
+  const EGLint impeller_config_attributes_no_msaa[] = {
+      EGL_RED_SIZE,   8, EGL_GREEN_SIZE, 8, EGL_BLUE_SIZE,    8,
+      EGL_ALPHA_SIZE, 8, EGL_DEPTH_SIZE, 0, EGL_STENCIL_SIZE, 8,
+      EGL_NONE};
+
+  EGLBoolean result;
+  EGLint num_config = 0;
+
   if (enable_impeller) {
     // First try the MSAA configuration.
-    if ((eglChooseConfig(egl_display_, impeller_config_attributes, &egl_config_,
-                         1, &numConfigs) == EGL_FALSE) ||
-        (numConfigs == 0)) {
-      // Next fall back to disabled MSAA.
-      if ((eglChooseConfig(egl_display_, impeller_config_attributes_no_msaa,
-                           &egl_config_, 1, &numConfigs) == EGL_FALSE) ||
-          (numConfigs == 0)) {
-        LogEglError("Failed to choose first context");
-        return false;
-      }
+    result = ::eglChooseConfig(display_, impeller_config_attributes, &config_,
+                               1, &num_config);
+
+    if (result == EGL_TRUE && num_config > 0) {
```

Comment:
Is `== EGL_TRUE` preferable to testing `result` outright?

Reply:
As in something like `if (result) { ... }`? Yes this is preferrable as EGL uses non-zero values to represent error codes. For example, EGL would return `0x300B` to present a context lost error.